File: backend/internal/models/vendor.go
package models

type Vendor struct {
	ID            int     `json:"id"`
	Name          string  `json:"name" validate:"required,min=2,max=255"`
	ContactPerson *string `json:"contact_person,omitempty"`
	Email         *string `json:"email,omitempty" validate:"omitempty,email"`
	Phone         *string `json:"phone,omitempty"`
	Address       *string `json:"address,omitempty"`
}

File: frontend/lib/models/vendor.dart
import 'package:json_annotation/json_annotation.dart';

part 'vendor.g.dart';

@JsonSerializable()
class Vendor {
  final int id;
  final String name;
  @JsonKey(name: 'contact_person')
  final String? contactPerson;
  final String? email;
  final String? phone;
  final String? address;

  Vendor({
    required this.id,
    required this.name,
    this.contactPerson,
    this.email,
    this.phone,
    this.address,
  });

  factory Vendor.fromJson(Map<String, dynamic> json) => _$VendorFromJson(json);
  Map<String, dynamic> toJson() => _$VendorToJson(this);
}
